### ISSUE TITLE
wrong unit conversion in MonteCarloMembraneBarostat.getSurfaceTension

### ISSUE THREAD

**Issue by Olllom** [openmm/openmm#2406]
The surface tension gets multiplied by 1e-25 when setting and getting the surface tension in units.

Part of the problem might be that the comment in MonteCarloMembraneBarostat.h specifies the unit as `bar` rather than `bar*nm` but I don't see how this would give a factor of 1e-25.

```
In [1]: from simtk import unit                                                                                                         

In [2]: from simtk.openmm import MonteCarloMembraneBarostat      

In [5]: barostat = MonteCarloMembraneBarostat(1.0*unit.bar, 1.0*unit.bar*unit.nanometer, 1, 1, 25)                                     

In [6]: barostat.getDefaultPressure()                                                                                                  
Out[6]: Quantity(value=1.0, unit=bar)

In [7]: barostat.getDefaultSurfaceTension()                                                                                            
Out[7]: Quantity(value=1e-25, unit=nanometer*bar)

In [8]: barostat = MonteCarloMembraneBarostat(1.0, 1.0, 1, 1, 25)                                                                      

In [9]: barostat.getDefaultPressure()                                                                                                  
Out[9]: Quantity(value=1.0, unit=bar)

In [10]: barostat.getDefaultSurfaceTension()                                                                                           
Out[10]: Quantity(value=1.0, unit=nanometer*bar)
```

**Comment by peastman** [comment]
Thanks!  Yes, you're right, it seems to be converting that incorrectly.  The Python API strips units from arguments by converting them to the standard "MD unit system".  And that leads to it getting multiplied by 1e-25:

```python
>>> (1.0*bar).value_in_unit_system(md_unit_system)
1e-25
```

That's happening because pressure is a derived unit (force per area) rather than a fundamental one.  So it expresses it in terms of the corresponding fundamental units:

```
>>> (1*bar).in_units_of(dalton*mole/nanometer/picosecond**2)
Quantity(value=1e-25, unit=mole*dalton/(nanometer*picosecond**2))
```

But the API expects pressure to be in bar, not mole*Da/nm/ps^2.  So the unit stripping code has special case handling for pressure: https://github.com/openmm/openmm/blob/master/wrappers/python/src/swig_doxygen/swig_lib/python/typemaps.i#L89.  But that doesn't catch the surface tension case.

I need to think a bit about the best way of handling this.

**Comment by jchodera** [comment]
That reminds me that we should really find a way to implement https://github.com/openmm/openmm/issues/204

**Comment by peastman** [comment]
Agreed, that does seem like the best long term solution.  It could be made optional: if units are specified they'll get checked, and if not it would keep the current behavior.

**Comment by peastman** [comment]
Although that still wouldn't completely solve this problem.  It would work for setting the default surface tension, but you'd still get the error if you tried to change the surface tension in the middle of a simulation by calling

```python
context.setParameter(MonteCarloMembraneBarostat::SurfaceTension(), 1*bar*nanometer)
```

**Comment by jchodera** [comment]
Handling that case appropriately would indeed seem to require a modern overhaul of the Python API, which could be timely.

**Comment by peastman** [comment]
No, I mean even if we overhauled it as described in #204, it *still* wouldn't handle that case correctly.  Admittedly it's a much less important case.  How often does anyone change the surface tension in the middle of a simulation?  But I thought I should point it out.

Ultimately the problem is that we're using inconsistent units.  If our unit system was fully consistent, pressures would be represented in units of 10^25 bar.

**Comment by peastman** [comment]
The fix is in #2411.
